src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessTest.cs

```diff
@@ -2,581 +2,428 @@
+
+                // Same test, this time explicitly set the property to false
+                Process p = CreateProcessInfinite();
+                p.EnableRaisingEvents = false;
+                p.Exited += delegate { isExitedInvoked = true; }; ;
+                StartAndKillProcessWithDelay(p);
+                Assert.False(isExitedInvoked, String.Format("Process_CanRaiseEvents0003: {0}", "isExited Event called with the EnableRaiseEvents = false"));
             }
-            
-            Assert.True(p.HasExited, "Process_HasExited003 failed");
         }
 
         [Fact]
-        public static void Process_MachineName()
+        public void Process_ExitCode()
         {
-            // Checking that the MachineName returns some value.
-            Process p = CreateProcessInfinite();
-            p.Start();
-            try
             {
-                Console.WriteLine(p.MachineName);
+                Process p = CreateProcess();
+                p.Start();
+                p.WaitForExit();
+                Assert.Equal(p.ExitCode, 100);
             }
-            finally
+
             {
-                p.Kill();
-                p.WaitForExit();
+                Process p = CreateProcessInfinite();
+                StartAndKillProcessWithDelay(p);
+                Assert.True(p.ExitCode < 0, String.Format("Process_ExitCode: Unexpected Exit code {0}", p.ExitCode));
             }
         }
 
         [Fact]
-        public static void Process_MainModule()
+        public void Process_ExitTime()
         {
-            // Get MainModule property from a Process object
             Process p = CreateProcessInfinite();
-            p.Start();
-            try
-            {
-                ProcessModule pMainModule = p.MainModule;
-                Assert.Equal(s_ProcessName, pMainModule.ModuleName);
-                Assert.True(pMainModule.FileName.Contains(pMainModule.ModuleName), "MainModule FileName invalid. " + GetModuleDescription(pMainModule));
-
-                // Check that the mainModule is present in the modules list.
-                ProcessModuleCollection allModules = p.Modules;
-                bool foundMainModule = false;
-                foreach (ProcessModule pModule in allModules)
-                {
-                    if ((pModule.BaseAddress == pMainModule.BaseAddress) && pModule.FileName.Equals(pMainModule.FileName))
-                    {
-                        foundMainModule = true;
-                        break;
-                    }
-                }
-                Assert.True(foundMainModule, "MainModule incorrect: " + GetModuleDescription(pMainModule) + Environment.NewLine + GetModulesDescription(allModules));
-            }
-            finally
-            {
-                p.Kill();
-                p.WaitForExit();
-            }
+            StartAndKillProcessWithDelay(p);
+            Assert.True(p.ExitTime < DateTime.Now, "Process_ExitTime is incorrect.");
```

Comment:
Is this necessarily true?  I'm wondering if it would depend on the granularity of the clock.